src/test/java/com/fasterxml/jackson/databind/deser/creators/AnySetterForCreator562Test.java

```diff
@@ -0,0 +1,180 @@
+package com.fasterxml.jackson.databind.deser.creators;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.InvalidDefinitionException;
+import com.fasterxml.jackson.databind.exc.MismatchedInputException;
+import com.fasterxml.jackson.databind.testutil.DatabindTestUtil;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+// [databind#562] Allow @JsonAnySetter on Creator constructors
+public class AnySetterForCreator562Test extends DatabindTestUtil
+{
+    static class POJO562
+    {
+        String a;
+        Map<String,Object> stuff;
+
+        @JsonCreator
+        public POJO562(@JsonProperty("a") String a,
+            @JsonAnySetter Map<String, Object> leftovers
+        ) {
+            this.a = a;
+            stuff = leftovers;
+        }
+    }
+
+    static class PojoWithNodeAnySetter
+    {
+        String a;
+        JsonNode anySetterNode;
+
+        @JsonCreator
+        public PojoWithNodeAnySetter(@JsonProperty("a") String a,
+            @JsonAnySetter JsonNode leftovers
+        ) {
+            this.a = a;
+            anySetterNode = leftovers;
+        }
+    }
+
+    static class MultipleAny562
+    {
+        @JsonCreator
+        public MultipleAny562(@JsonProperty("a") String a,
+            @JsonAnySetter Map<String, Object> leftovers,
+            @JsonAnySetter Map<String, Object> leftovers2) {
+            throw new Error("Should never get here!");
+        }
+    }
+
+    static class PojoWithDisabled
+    {
+        String a;
+        Map<String,Object> stuff;
+
+        @JsonCreator
+        public PojoWithDisabled(@JsonProperty("a") String a,
+            @JsonAnySetter(enabled = false) Map<String, Object> leftovers
+        ) {
+            this.a = a;
+            stuff = leftovers;
+        }
+    }
+
+    private final ObjectMapper MAPPER = newJsonMapper();
+
+    @Test
+    public void mapAnySetterViaCreator562() throws Exception
+    {
+        Map<String, Object> expected = new HashMap<>();
+        expected.put("b", Integer.valueOf(42));
+        expected.put("c", Integer.valueOf(111));
+
+        POJO562 pojo = MAPPER.readValue(a2q(
+                "{'a':'value', 'b':42, 'c': 111}"
+                ),
+                POJO562.class);
+
+        assertEquals("value", pojo.a);
+        assertEquals(expected, pojo.stuff);
+
+        // Should also initialize any-setter-Map even if no contents
```

Comment:
With change I made wrt ordering (handle buffered any props before null checking), an empty `Map` will be created for Any property for this case, no `null` should remain.